### ISSUE TITLE
✨ Create Communities pages

### ISSUE THREAD

**Issue by tristan-orourke** [GCTC-NTGC/gc-digital-talent#10361]
## ✨ Feature

Create pages where Communities can be created, can be edited, and Community Managers and Community Recruiters can be added.

## 🕵️ Details

These pages can basically be a reflavour of the Teams pages.

Who can access these pages:
- Community Admin, Community Recruiter, and Platform admin can see the Communities table, and the info page for a specific community (which lists all team members and their roles)
- Only Platform Admins can create a new Community
- Platform Admin and Community Admin can update an existing community.
- Platform Admin can give and take Community Admin roles
- Community Admin and Community Recruiter can give and take Community Recruiter roles

Create new graphql queries and mutations as necessary for these pages to work.

For users with the appropriate roles, ensure the appropriate links appear on the Admin Dashboard page and the nav menu.

## 🎨 Design File

[Figma file](https://www.figma.com/design/3VPSAv5689GK51Us6BxbJm/Admin---Communities?node-id=10092-9691&t=qHufZj1JvAkKY1wD-1) by @Jerryescandon 

## 📸 Screenshot

<details><summary>📸 Table</summary>
<p>

![COMMUNITY TABLE-2](https://github.com/GCTC-NTGC/gc-digital-talent/assets/16977254/6a0bfdb0-ebe7-4a30-96ed-a89eff18cd5e)

</p>
</details> 
<details><summary>📸 Create a new community</summary>
<p>

![COMMUNITY TABLE](https://github.com/GCTC-NTGC/gc-digital-talent/assets/16977254/1524fa96-c79f-41ce-8d64-cb311edca443)

</p>
</details> 

<details><summary>📸 Information Tab</summary>
<p>

![COMMUNITY TABLE](https://github.com/GCTC-NTGC/gc-digital-talent/assets/16977254/72d78eb9-1dcc-4b6b-8540-5a745a63005d)

![COMMUNITY TABLE](https://github.com/GCTC-NTGC/gc-digital-talent/assets/16977254/22165240-35d7-441b-bc8d-0f7fd129cbcb)

</p>
</details> 
<details><summary>📸 Manage Access Tab</summary>
<p>

![COMMUNITY TABLE-1](https://github.com/GCTC-NTGC/gc-digital-talent/assets/16977254/e12ac29e-487d-4b81-8d0e-300d1c0cd20e)

</p>
</details> 

<details><summary>📸 Modals</summary>
<p>

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/16977254/a16b4667-7eae-4dbc-a378-c11f29b889a3)

</p>
</details> 

<details><summary>📸 Dashboard and menu buttons</summary>
<p>

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/f03f93d8-2f77-4295-aed5-0bdc470c091b)

</p>
</details> 

## 🙋‍♀️ Proposed Implementation

For assigning/removing roles, use the mutation deleveloped in #10351.

GraphQL details TBD

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Platform Admins can view, create, and update Communities and their team members
- [ ] Community Admins and Community Recruiters can view the Communities they're in, and their team members
- [ ] Community Admins can edit Community details and assign Community Recruiters
- [ ] Community pages can be easily accessed from the Admin Dashboard and nav menu for users with these roles
- [ ] phpunit tests for community mutations
- [ ] playwright tests for new pages?
- [ ] Use the following urls:
    - `/admin/communities`
    - `/admin/communities/create`
    - `/admin/communities/:id`
    - `/admin/communities/:id/manage-access`

OUT OF SCOPE: 
- Do not build archiving and restoring communities functionality yet.
- Do not remove the teams buttons from the dashboard page yet.

## 🛑 Blockers

Issues which must be completed before this one.

```[tasklist]
### Blocked By
- [ ] #10351
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/10349
```

## Deployment
Use the new UI to assign users who currently have to Community Manager or Request Responder roles to Community Admin and Community Recruiter in the appropriate community. (Consult @gobyrne )

**Comment by petertgiles** [comment]
Let's nail down URLs, too, in the design process.

**Comment by petertgiles** [comment]
I will merge with https://github.com/GCTC-NTGC/gc-digital-talent/issues/10442

**Comment by mnigh** [comment]
@tristan-orourke a few questions related to roles:

> Community Admins and Community Recruiters can view the Communities they're in, and their team members
1. can a user with either of those roles view communities they are **not** in?

> Community Admins can edit Community details and assign Community Recruiters
2. can a user with a Community Admin role edit community details for communities that they do **not** belong to? can a user with a Community Admin role manage members for communities that they do **not** belong to?

3. if "Maintaining the platform" is for `platform_admin`and "Publishing pools and managing teams" is for `community_manager`, should `community_admin` and `community_recruiter` have their own admin dashboard section with a button for Communities?

**Comment by tristan-orourke** [comment]
1. Community details are public information, however, Community Admins and Community Recruiters cannot see the _members_ of other communities.
2. A Community Admin cannot edit the details or members of a community they are not an Admin for.
3. Don't worry about adding a new section right now, just add a Communities link next to the Teams links.
